File: packages/mobile/app/src/main/java/com/devhub/devhubapp/api/NotificationAPI.kt
package com.devhub.devhubapp.api

import com.devhub.devhubapp.dataClasses.NotificationResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.PATCH

interface NotificationAPI {
    @GET("notifications/{user_id}")
    fun getNotifications(@Path("user_id") userId: String): Call<NotificationResponse>

    @PATCH("notifications/{notification_id}")
    fun markAsRead(@Path("notification_id") notificationId: String): Call<Map<String, String>>
}

File: packages/web/src/api/notification.api.ts
import { createApi } from '@reduxjs/toolkit/query/react'

import baseQueryWithReauth from './baseQueryWithReauth'

import { GetNotificationsRes } from '~types/notifications/notifications.types'

export const api = createApi({
  reducerPath: 'notificationApi',
  baseQuery: baseQueryWithReauth,
  endpoints: builder => ({
    getNotificationsByUser: builder.query<GetNotificationsRes, { user_id: string | undefined }>({
      query: ({ user_id }) => ({
        url: `notifications/${user_id}`,
        method: 'GET',
      }),
    }),
    readNotificationById: builder.mutation<{ message: string }, { notification_id: string | undefined }>({
      query: ({ notification_id }) => ({
        url: `notifications/${notification_id}`,
        method: 'PATCH',
      }),
    }),
  }),
})

export const { useGetNotificationsByUserQuery, useReadNotificationByIdMutation, useLazyGetNotificationsByUserQuery } =
  api
